engine/apps/integrations/views.py

```diff
@@ -391,3 +391,39 @@ def _process_heartbeat_signal(self, request, alert_receive_channel):
         process_heartbeat_task.apply_async(
             (alert_receive_channel.pk,),
         )
+
+
+class AlertManagerV2View(BrowsableInstructionMixin, AlertChannelDefiningMixin, IntegrationRateLimitMixin, APIView):
+    """
+    AlertManagerV2View consumes alerts from AlertManager. It expects data to be in format of AM webhook receiver.
+    """
+
+    def post(self, request, *args, **kwargs):
+        alert_receive_channel = self.request.alert_receive_channel
+        if not alert_receive_channel.integration == AlertReceiveChannel.INTEGRATION_ALERTMANAGER_V2:
+            return HttpResponseBadRequest(
+                f"This url is for integration with {alert_receive_channel.config.title}."
+                f"Key is for {alert_receive_channel.get_integration_display()}"
+            )
+        alerts = request.data.get("alerts", [])
+
+        data = request.data
+        if "numFiring" not in request.data:
+            # Count firing and resolved alerts manually if not present in payload
+            num_firing = len(list(filter(lambda a: a["status"] == "firing", alerts)))
+            num_resolved = len(list(filter(lambda a: a["status"] == "resolved", alerts)))
+            data = {**request.data, "numFiring": num_firing, "numResolved": num_resolved}
+
+        create_alert.apply_async(
```

Comment:
technically we save alertmanager alert group as oncall alert. Need to understand how to represent that correctly

Reply:
I think it looks fine in OnCall terminology